SimpleExec/ProcessExtensions.cs

```diff
@@ -12,12 +14,31 @@ public static void Run(this Process process, bool noEcho, string echoPrefix)
             process.WaitForExit();
         }
 
-        public static Task RunAsync(this Process process, bool noEcho, string echoPrefix)
+        public static Task RunAsync(this Process process, bool noEcho, string echoPrefix, CancellationToken cancellationToken = default)
         {
             var tcs = new TaskCompletionSource<object>();
-            process.Exited += (s, e) => tcs.SetResult(default);
+            var cancelled = false;
+
+            process.Exited += (s, e) =>
+            {
+                if (cancelled)
+                {
+                    tcs.TrySetCanceled(cancellationToken);
+                }
+                else
+                {
+                    tcs.SetResult(default);
+                }
+            };
+
             process.EnableRaisingEvents = true;
             process.EchoAndStart(noEcho, echoPrefix);
+
+            cancellationToken.Register(() => {

```

Comment:
```suggestion
            var ctRegistration = cancellationToken.Register(() => {
```

(and move this up so you can dispose `ctRegistration` when setting result)